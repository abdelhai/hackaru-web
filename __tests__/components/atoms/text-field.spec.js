import Factory from '@/__tests__/__setups__/factory';
import TextField from '@/components/atoms/text-field';

describe('TextField', () => {
  let factory;
  let wrapper;

  beforeEach(() => {
    factory = new Factory(TextField, {
      propsData: {
        value: 'value'
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when enter text', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setValue('updated');
    });

    it('emit input', () => {
      expect(wrapper.emitted('input')).toBeTruthy();
    });
  });
});
