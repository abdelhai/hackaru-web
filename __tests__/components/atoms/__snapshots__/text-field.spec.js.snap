// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField render correctly 1`] = `<input />`;
